OfficeDev/office-ui-fabric-react
# Add red border when required TextField is empty and goes out of focus

#### Pull request checklist

- [x] Addresses an existing issue: Fixes #8213 

#### Description of changes
If it is a `required` field, set the color of the `TextField` to `red` on losing `focus`, and unset the `red` color when `focus` is regained. 

#### Focus areas to test
1. Have a required TextField
2. Place focus on the TextField
3. Click elsewhere leaving the TextField empty
4. The border of the TextField should now be `red`

![red-border](https://user-images.githubusercontent.com/8922176/54341374-06830800-45f7-11e9-866e-49b5b2be9198.gif)


###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8311)
